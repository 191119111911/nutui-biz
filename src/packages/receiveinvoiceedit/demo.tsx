import React, { useState } from "react";
import { ReceiveInvoiceEdit } from "./receiveinvoiceedit";
import { Radio, Cell, CellGroup } from "@nutui/nutui-react";
import { useTranslate } from "../../sites/assets/locale";
interface tarnslatedOption {
  basic: string;
  changeAddress: string;
  addAddress: string;
}

const ReceiveInvoiceEditDemo = () => {
  const [translated] = useTranslate<tarnslatedOption>({
    "zh-CN": {
      basic: "基本用法",
      changeAddress: "修改地址",
      addAddress: "新增地址",
    },
    "zh-TW": {
      basic: "基本用法",
      changeAddress: "修改地址",
      addAddress: "新增地址",
    },
    "en-US": {
      basic: "Basic usage",
      changeAddress: "Modify address",
      addAddress: "Add Address",
    },
  });

  const addressData: any = {
    addressSelect: [],
    addressTitle: "选择所在地区",
    province: [
      { id: 1, name: "北京", title: "B" },
      { id: 2, name: "广西", title: "G" },
      { id: 3, name: "江西", title: "J" },
      { id: 4, name: "四川", title: "S" },
      { id: 5, name: "浙江", title: "Z" },
    ],
    city: [
      { id: 7, name: "朝阳区", title: "C" },
      { id: 8, name: "崇文区", title: "C" },
      { id: 9, name: "昌平区", title: "C" },
      { id: 6, name: "石景山区", title: "S" },
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 10, name: "北苑", title: "B" },
    ],
    country: [
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 9, name: "北苑", title: "B" },
      { id: 4, name: "常营乡", title: "C" },
    ],
    town: [],
  };
  const invoiceInfo = {
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
  };
  const invoiceInfo2 = {
    name: "xx",
    tel: "12345678913",
    region: "北京朝阳区八里庄街道",
    regionIds: [1, 7, 3],
    address: "xxx小区3-2-302",
  };
  const addressSetData = {
    required: ["name", "tel"],
  };

  const addressData2: any = {
    addressSelect: [1, 7, 3],
    addressTitle: "选择所在地区",
    province: [
      { id: 1, name: "北京", title: "B" },
      { id: 2, name: "广西", title: "G" },
      { id: 3, name: "江西", title: "J" },
      { id: 4, name: "四川", title: "S" },
      { id: 5, name: "浙江", title: "Z" },
    ],
    city: [
      { id: 7, name: "朝阳区", title: "C" },
      { id: 8, name: "崇文区", title: "C" },
      { id: 9, name: "昌平区", title: "C" },
      { id: 6, name: "石景山区", title: "S" },
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 10, name: "北苑", title: "B" },
    ],
    country: [
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 9, name: "北苑", title: "B" },
      { id: 4, name: "常营乡", title: "C" },
    ],
    town: [],
  };

  const onChange = (data: any) => {
    console.log("onChangeAddress", data);
  };

  const onClose = (data: any) => {
    console.log("onCloseAddress", data);
  };


  return (
    <div className="demo">
      <h2>{translated.basic}</h2>
      <ReceiveInvoiceEdit
        address={addressData}
        data={addressSetData}
        invoiceInfo={invoiceInfo}
        onSave={(formData) => {
          console.log(formData);
        }}
      />
      <h2>{translated.changeAddress}</h2>
      <ReceiveInvoiceEdit
        address={addressData2}
        data={addressSetData}
        invoiceInfo={invoiceInfo2}
        onSave={(formData) => {
          console.log(formData);
        }}
        onAddressChange={onChange}
        onAddressClose={onClose}
      />
    </div>
  );
};

export default ReceiveInvoiceEditDemo;
