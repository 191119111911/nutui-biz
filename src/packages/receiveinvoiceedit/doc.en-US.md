# ReceiveInvoiceEdit Edit receipt address

### Intro

Billing address editing, commonly found on the billing address management page, is mainly used to add or edit or modify addresses.

### Install

```javascript
import { ReceiveInvoiceEdit } from '@nutui/nutui-react';
```

## Demo

### Basic usage

:::demo

```tsx
import React from 'react';
import { ReceiveInvoiceEdit } from '@nutui/nutui-biz';

const App = () => {
     const addressData: any = {
     addressSelect: [],
     addressTitle: "Select your region",
     province: [
       { id: 1, name: "Beijing", title: "B" },
       { id: 2, name: "Guangxi", title: "G" },
       { id: 3, name: "Jiangxi", title: "J" },
       { id: 4, name: "Sichuan", title: "S" },
       { id: 5, name: "Zhejiang", title: "Z" },
     ],
     city: [
       { id: 7, name: "Chaoyang District", title: "C" },
       { id: 8, name: "Chongwen District", title: "C" },
       { id: 9, name: "Changping District", title: "C" },
       { id: 6, name: "Shijingshan District", title: "S" },
       { id: 3, name: "Balizhuang Street", title: "B" },
       { id: 10, name: "Beiyuan", title: "B" },
     ],
     country: [
       { id: 3, name: "Balizhuang Street", title: "B" },
       { id: 9, name: "Beiyuan", title: "B" },
       { id: 4, name: "Changying Township", title: "C" },
     ],
     town: [],
   };
   const invoiceInfo = {
     name: "",
     tel: "",
     region: "",
     regionIds: [],
     address: "",
   };
   const addressSetData = {
     required: ["name", "tel"],
   };
   return (
     <ReceiveInvoiceEdit
         address={addressData}
         data={addressSetData}
         invoiceInfo = {invoiceInfo}
         onSave={(formData) => {
             console.log(formData);
         }}
         />
   );
};
export default App;
```
:::
### Modification of existing address information

:::demo

```tsx
import React from 'react';
import { ReceiveInvoiceEdit } from '@nutui/nutui-biz';

const App = () => {
   const addressData: any = {
     addressSelect: [1, 7, 3],
     addressTitle: "Select your region",
     province: [
     { id: 1, name: "Beijing", title: "B" },
     { id: 2, name: "Guangxi", title: "G" },
     { id: 3, name: "Jiangxi", title: "J" },
     { id: 4, name: "Sichuan", title: "S" },
     { id: 5, name: "Zhejiang", title: "Z" },
     ],
     city: [
     { id: 7, name: "Chaoyang District", title: "C" },
     { id: 8, name: "Chongwen District", title: "C" },
     { id: 9, name: "Changping District", title: "C" },
     { id: 6, name: "Shijingshan District", title: "S" },
     { id: 3, name: "Balizhuang Street", title: "B" },
     { id: 10, name: "Beiyuan", title: "B" },
     ],
     country: [
     { id: 3, name: "Balizhuang Street", title: "B" },
     { id: 9, name: "Beiyuan", title: "B" },
     { id: 4, name: "Changying Township", title: "C" },
     ],
     town: [],
   };
   const invoiceInfo = {
     name: "Zhang San",
     tel: "13141234567",
     region: "Balizhuang Street, Chaoyang District, Beijing",
     regionIds: [1, 7, 3],
     address: "xxx community 3-2-302",
   };
   const addressSetData = {
     required: ["name", "tel"],
   };

   const onChange = (data: any) => {
     console.log("onAddressChange", data);
   };

   const onClose = (data: any) => {
     console.log("onAddressClose", data);
   };

   return (
     <ReceiveInvoiceEdit
         address={addressData}
         data={addressSetData}
         invoiceInfo = {invoiceInfo}
         onSave={(formData) => {
             console.log(formData);
         }}
         onAddressChange={onChange}
         onAddressClose = {onClose}
         />
   );
};
export default App;
```
:::





## API

### Props


| Field | Description | Type | Default |
|---------|--------------------------------------- -----|---------|-----------|
| address | Information about address components | Object | {} |
| data | Edit billing address data format settings | Object | {} |
| invoiceInfo | invoice address value | Object |{} |

### Props address
| Field | Description | Type | Default |
|---------|--------------------------------------- -----|---------|-----------|
| addressSelect | set default selected address | String\|Number[] | [] |
| addressStr | address information | String | '' |
| province | province, each province object must have a name field, if the type is custom2, the title field must be specified as the first letter | Array | [] |
| city | city, each city object must have a name field, if the type is custom2, the title field must be specified as the first letter | Array | [] |
| country | County, each county object must have a name field, if the type is custom2, the title field must be specified as the first letter | Array | [] |
| town | township/township, each township/town object must have a name field, if the type is custom2, the title field must be specified as the first letter | Array | [] |
| addressTitle | Custom address selection text | String | 'Select your region' |

### Props data
| Field | Description | Type | Default |
|---------|--------------------------------------- -----|---------|-----------|
| nameText | Custom consignee text | String | 'Consignee' |
| namePlaceholder | Custom consignee placeholder text | String | 'Please enter the consignee' |
| nameErrorMsg | Custom consignee non-null verification error prompt text | String | 'This item is required, please fill it out and submit it' |
| telText | Custom mobile phone number text | String | 'Mobile phone number' |
| telPlaceholder | Custom phone number placeholder text | String | 'Please enter your phone number' |
| telErrorMsg | Prompt text for non-null mobile phone number verification error | String | 'This item is required, please fill it out and submit it' |
| regionText | Customize the region text | String | 'Region' |
| regionPlaceholder | Custom region placeholder text | String | 'Please select your region' |
| regionErrorMsg | Custom region non-null verification error prompt text | String | 'This item is required, please fill it out and submit it' |
| addressText | Custom detailed address text | String | 'Detailed address' |
| addressPlaceholder | Custom detailed address placeholder text | String | 'street, building number' |
| addressErrorMsg | Custom detailed address non-null verification error prompt text | String | 'This item is required, please fill it out and submit it' |
| required |Fields that need to display error messages | String[] | ['name','tel'] |
| bottomText | Customize the text of the save button at the bottom | String | 'Save' |

### Props addressInfo
| Field | Description | Type | Default |
|---------|--------------------------------------- -----|---------|-----------|
| name | consignee information | String | '' |
| tel | phone number information | String | '' |
| region | region information | String | '' |
| regionIds | The id information of the selected address in the region | String | '' |
| address | detailed address information | String | '' |



### Events
| Field | Description | Callback Parameters |
|----- | ----- | ----- |
| onChange | Enter text in the input box, return the input text and the tag of the input box where it is located tag (name,tel,region,address) | value,tag |
| onAddressChange | Triggered when a region is selected when customizing an address | {custom: the currently selected address, next: the next-level address, value: the currently selected address information} |
| onAddressClose | Triggered when the address selection dialog is closed | value |
| onSave | Click the save address button at the bottom to return the saved information | value |