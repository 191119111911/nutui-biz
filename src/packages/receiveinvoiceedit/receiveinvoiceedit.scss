.nb-receive-invoice-edit{
    background: $white;
    padding:5px 10px;
    border-radius: 4px;
    &__item{
        background: $white;
        padding: 0 10px;
        font-size: $font-size-2;
        color:$title-color2;
    }
    &__bottom{
        bottom: constant(safe-area-inset-bottom);
        bottom: env(safe-area-inset-bottom);
        width: 100%;
        padding: 6px 15px;
        background-color: #fff;
    }
    .nut-input{
        padding: var(--nutui-input-padding, 10px 10px);
    }
    .nut-input.nut-input-required:before{
        left: 0;
    }
}