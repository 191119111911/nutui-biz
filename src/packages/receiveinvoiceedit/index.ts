import { ReceiveInvoiceEdit } from './receiveinvoiceedit'

export default ReceiveInvoiceEdit
