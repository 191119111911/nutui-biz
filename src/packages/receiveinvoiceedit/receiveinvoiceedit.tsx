import React, { FunctionComponent, useState, useEffect } from "react";
import { IComponent } from "@/utils/typings";
import {
  Input,
  Button,
  Address,
} from "@nutui/nutui-react";
import "./receiveinvoiceedit.scss";
import { useConfig } from "@/packages/configprovider";

interface CalResult {
  type: string;
  data: any;
}
interface RegionData {
  name?: string;
  [key: string]: any;
}
interface InvoiceInfo {
  name?: string;
  tel?: string;
  region?: string;
  regionIds?: any;
  address?: string;
}
interface InvoiceData {
  nameText?: string;
  namePlaceholder?: string;
  nameErrorMsg?: string;
  telText?: string;
  telPlaceholder?: string;
  telErrorMsg?: string;
  regionText?: string;
  regionPlaceholder?: string;
  regionErrorMsg?: string;
  addressText?: string;
  addressPlaceholder?: string;
  addressErrorMsg?: string;
  required?: any;
  bottomText?: string;
}
interface AddressResult {
  addressSelect: any;
  addressStr: string;
  province: RegionData[];
  city: RegionData[];
  country: RegionData[];
  town: RegionData[];
  addressTitle: string;
}

export interface ReceiveInvoiceEdit extends IComponent {
  invoiceInfo: InvoiceInfo;
  data: InvoiceData;
  address: AddressResult;
  onChange: (val: string, tag: string) => void;
  onChangeAddress: (data: any) => void;
  onCloseAddress: (data: any) => void;
  onSave: (data: any) => void;
}

const defaultProps = {
  invoiceInfo: {
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
  },
  data: {},
  address: {
    modelValue: false,
    addressSelect: [],
    province: [],
    city: [],
    country: [],
    town: [],
    addressTitle: "选择所在地区",
  },
} as unknown as ReceiveInvoiceEdit;

export const ReceiveInvoiceEdit: FunctionComponent<
  Partial<ReceiveInvoiceEdit> &
    Omit<React.HTMLAttributes<HTMLDivElement>, "onChange">
> = (props) => {
  const { locale } = useConfig();
  const {
    invoiceInfo = null,
    showSave = false,
    data = null,
    address,
    onChange,
    onSave,
    onChangeAddress,
    onCloseAddress,
    ...rest
  } = {
    ...props,
  };
  //提交数据格式
  const [formData, setFormData] = useState<any>({
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
  });
  //必填项设置
  const [isRequired, setIsRequired] = useState([
    "name",
    "tel",
    "region",
    "address",
  ]);
  //地址组件相关数据
  const [addressData, setAddressData] = useState({
    addressSelect: [],
    province: [] as RegionData[],
    city: [] as RegionData[],
    country: [] as RegionData[],
    town: [] as RegionData[],
    addressTitle: "选择所在地区",
  });
  //地址编辑数据形式兜底文案配置
  const [editSeting, setEditSeting] = useState({
    nameText: locale.receiveInvoiceEdit.nameText,
    namePlaceholder: locale.receiveInvoiceEdit.namePlaceholder,
    nameErrorMsg: locale.receiveInvoiceEdit.nameErrorMsg,
    telText: locale.receiveInvoiceEdit.telText,
    telPlaceholder: locale.receiveInvoiceEdit.telPlaceholder,
    telErrorMsg: locale.receiveInvoiceEdit.telErrorMsg,
    regionText: locale.receiveInvoiceEdit.regionText,
    regionPlaceholder: locale.receiveInvoiceEdit.regionPlaceholder,
    regionErrorMsg: locale.receiveInvoiceEdit.regionErrorMsg,
    addressText: locale.receiveInvoiceEdit.addressText,
    addressPlaceholder: locale.receiveInvoiceEdit.addressPlaceholder,
    addressErrorMsg: locale.receiveInvoiceEdit.addressErrorMsg,
    bottomText: locale.receiveInvoiceEdit.bottomText,
  });
  const [errorList, setErrorList] = useState<any>([]);
  const [addressVisible, setAddressVisible] = useState(false);

  useEffect(() => {
    //地址组件所需相关信息获取
    if (address) {
      setAddressData(address);
    }
    //编辑地址所需字段设置获取
    if (data) {
      setEditSeting({ ...editSeting,...data });
      if (data?.required) {
        setIsRequired(data.required);
      }
    }
    //地址信息初始化获取
    if (invoiceInfo) {
      setFormData({ ...formData, ...invoiceInfo });
      //同步地址组件
      if (invoiceInfo.regionIds.length) {
        setAddressData({
          ...addressData,
          addressSelect: invoiceInfo.regionIds,
        });
      }
    }
    //重置error错误
    setErrorList([]);
  }, [address, data, invoiceInfo]);
  const changeAddress = (cal: any) => {
    setErrorList(errorList.filter((i: any) => i != "region"));
    if (cal.next === "town") {
      setAddressVisible(false);
    }
    onChangeAddress && onChangeAddress(cal);
  };

  const closeAddress = (val: CalResult) => {
    if ((val.data as AddressResult).addressStr) {
      //地址id格式处理
      let ids = val.data.addressIdStr.split("_").map((item: any) => item * 1);
      //删除数组中为0的项
      ids.splice(ids.indexOf(0));
      //提交信息数据同步
      setFormData({ ...formData, region: val.data.addressStr, regionIds: ids });
      //地址组件数据同步
      setAddressData({ ...addressData, addressSelect: ids });
    }
    onCloseAddress && onCloseAddress(val);

    setAddressVisible(false);
  };
  const inputOnchange = (val: any, tag: string) => {
    let data = { ...formData };
    if (val.length != 0) {
      setErrorList(errorList.filter((i: any) => i != tag));
    }

    Object.keys(formData).map((key: any) => {
      if (key === tag) {
        if (key === "tel") {
          const regTel =
            /^(13[0-9]|14[01456879]|15[0-35-9]|16[2567]|17[0-8]|18[0-9]|19[0-35-9])\d{8}$/;
          const regNumber = /[^-0-9]/g;
          let number =
            !regTel.test(val) && val.length > 11
              ? val.substring(0, 11)
              : val.replace(regNumber, "");
          data[tag] = number;
        } else {
          data[tag] = val;
        }
      }
    });
    setFormData({ ...formData, ...data });
    onChange && onChange(val, tag);
  };
  const validForm = () => {
    let form = formData;
    let flag = false
    let arr: any = [].concat(errorList);
    Object.keys(form).map((key) => {
      // console.log(key);
      // console.log(isRequired);
      if (isRequired.includes(key) && formData[key] == "") {
        console.log(key);
        
        switch (key) {
          case "name":
            if (!errorList.includes("name")) {
              arr.push("name");
              setErrorList(arr);
            }
            break;
          case "tel":
            if (!errorList.includes("tel")) {
              arr.push("tel");
              setErrorList(arr);
            }
            break;
          case "region":
            if (!errorList.includes("region")) {
              arr.push("region");
              setErrorList(arr);
            }
            break;
          case "address":
            if (!errorList.includes("address")) {
              arr.push("address");
              setErrorList(arr);
            }
            break;

          default:
            break;
        }
        flag = false
      }else{
        flag =  true
      }
    });
    return flag
  };
  const save = () => {
    if (validForm()) {
      onSave && onSave(formData);
    } else {
      console.log("校验不通过");
    }
  };

  return (
    <div className="nut-receive-invoice-edit" {...rest}>
      <div className="nut-receive-invoice-edit-item">
        <Input
          className="nut-input-text"
          label={editSeting.nameText}
          defaultValue={formData.name}
          name="name"
          placeholder={editSeting?.namePlaceholder}
          type="text"
          required={isRequired.includes("name") || false}
          onChange={(e) => inputOnchange(e, "name")}
          errorMessage={errorList.includes("name") && editSeting.nameErrorMsg}
        />
      </div>
      <div className="nut-receive-invoice-edit-item">
        <Input
          className="nut-input-text"
          label={editSeting.telText}
          defaultValue={formData.tel}
          name="tel"
          placeholder={editSeting.telPlaceholder}
          type="tel"
          required={isRequired.includes("tel") || false}
          onChange={(e) => inputOnchange(e, "tel")}
          errorMessage={errorList.includes("tel") && editSeting.telErrorMsg}
        />
      </div>
      <div className="nut-receive-invoice-edit-item">
        <Input
          className="nut-input-text"
          label={editSeting.regionText}
          defaultValue={formData.region}
          name="region"
          readonly
          placeholder={editSeting.regionPlaceholder}
          type="text"
          required={isRequired.includes("region") || false}
          onClick={() => setAddressVisible(true)}
          errorMessage={
            errorList.includes("region") && editSeting.regionErrorMsg
          }
        />
        <Address
          modelValue={addressVisible}
          modelSelect={addressData.addressSelect}
          province={addressData.province}
          city={addressData.city}
          country={addressData.country}
          town={addressData.town}
          customAddressTitle={addressData.addressTitle}
          onChange={(cal) => changeAddress(cal)}
          onClose={closeAddress}
        />
      </div>
      <div className="nut-receive-invoice-edit-item">
        <Input
          label={editSeting.addressText}
          className="nut-input-text"
          defaultValue={formData.address}
          placeholder={editSeting.addressPlaceholder}
          type="text"
          required={isRequired.includes("address") || false}
          onChange={(e) => inputOnchange(e, "address")}
          errorMessage={
            errorList.includes("address") && editSeting.addressErrorMsg
          }
        />
      </div>
      <div className="nut-receive-invoice-edit-bottom">
        <Button block type="danger" onClick={save}>
          {editSeting.bottomText}
        </Button>
      </div>
    </div>
  );
};

ReceiveInvoiceEdit.defaultProps = defaultProps;
ReceiveInvoiceEdit.displayName = "NutReceiveInvoiceEdit";
