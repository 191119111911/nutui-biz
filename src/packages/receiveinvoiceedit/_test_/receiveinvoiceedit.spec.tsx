import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { ReceiveInvoiceEdit } from "../receiveinvoiceedit";

const addressData: any = {
  addressSelect: [],
  addressTitle: "选择所在地区",
  province: [
    { id: 1, name: "北京", title: "B" },
    { id: 2, name: "广西", title: "G" },
    { id: 3, name: "江西", title: "J" },
    { id: 4, name: "四川", title: "S" },
    { id: 5, name: "浙江", title: "Z" },
  ],
  city: [
    { id: 7, name: "朝阳区", title: "C" },
    { id: 8, name: "崇文区", title: "C" },
    { id: 9, name: "昌平区", title: "C" },
    { id: 6, name: "石景山区", title: "S" },
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 10, name: "北苑", title: "B" },
  ],
  country: [
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 9, name: "北苑", title: "B" },
    { id: 4, name: "常营乡", title: "C" },
  ],
  town: [],
};
const invoiceInfo = {
  name: "",
  tel: "",
  region: "",
  regionIds: [],
  address: "",
};
const addressSetData = {
  nameText: "收件人",
  namePlaceholder: "请输入收件人姓名",
  isRequired: ["name", "tel", "region", "address"],
  isDefualtAddress: true,
};
test("test props & event", async () => {

  const saveHandle = jest.fn();
  const { container } = render(
    <ReceiveInvoiceEdit
      address={addressData}
      data={addressSetData}
      invoiceInfo={invoiceInfo}
      onSave={saveHandle}
    />
  );
  const inputItem = container.querySelectorAll('.nut-receive-invoice-edit-item') 
  expect(inputItem).toBeTruthy()
  const requiredItem = container.querySelectorAll(".nut-input-required");
  await waitFor(() => {
    expect(requiredItem.length).toBe(4);
    expect(container.querySelectorAll(".label-string")[0].innerHTML).toBe(
      addressSetData.nameText
    );
    expect(container).toMatchSnapshot();
  });
  const bottomEl = container.querySelector(
    ".nut-receive-invoice-edit-bottom .nut-button"
  ) as HTMLElement;
  fireEvent.click(bottomEl);
  const inputEl = container.querySelectorAll(".nut-input-text")[0];
  const input = inputEl.querySelector('.input-text') as Element
  fireEvent.change(input, { target: { value: '123' } })
  setTimeout(() => {
    expect(input).toHaveAttribute(
      'value',
      '123'
    )
    fireEvent.blur(input)
  }, 300)
});
test("test props & event", async () => {
  const invoiceInfo2 = {
    name: "xx",
    tel: "12345678913",
    region: "北京朝阳区八里庄街道",
    regionIds: [1, 7, 3],
    address: "xxx小区3-2-302",
  };
  const addressSetData = {
    required: ["name", "tel"],
  };

  const addressData2: any = {
    addressSelect: [1, 7, 3],
    addressTitle: "选择所在地区",
    province: [
      { id: 1, name: "北京", title: "B" },
      { id: 2, name: "广西", title: "G" },
      { id: 3, name: "江西", title: "J" },
      { id: 4, name: "四川", title: "S" },
      { id: 5, name: "浙江", title: "Z" },
    ],
    city: [
      { id: 7, name: "朝阳区", title: "C" },
      { id: 8, name: "崇文区", title: "C" },
      { id: 9, name: "昌平区", title: "C" },
      { id: 6, name: "石景山区", title: "S" },
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 10, name: "北苑", title: "B" },
    ],
    country: [
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 9, name: "北苑", title: "B" },
      { id: 4, name: "常营乡", title: "C" },
    ],
    town: [],
  };
  const changeHandle = jest.fn();
  const closeHandle = jest.fn();
  const saveHandle = jest.fn();
  const { container } = render(
    <ReceiveInvoiceEdit
      address={addressData2}
      data={addressSetData}
      invoiceInfo={invoiceInfo2}
      onAddressChange={changeHandle}
      onAddressClose={closeHandle}
      onSave={saveHandle}
    />
  );

  await waitFor(() => {
    const inputEl = container.querySelectorAll(".nut-input-text")[2];
    inputEl && fireEvent.click(inputEl);
    const regionEl = container.querySelector(".nut-address ");
    expect(regionEl).toBeTruthy;

    const regionItem = regionEl?.querySelectorAll(
      ".nut-address__region-item"
    )[0];
    regionItem && fireEvent.click(regionItem);

    waitFor(() => {
      expect(changeHandle).toBeCalled();
      expect(closeHandle).toBeCalled();
    });
  });
});
