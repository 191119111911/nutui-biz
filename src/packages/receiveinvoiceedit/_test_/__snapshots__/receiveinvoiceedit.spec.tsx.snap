// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test props & event 1`] = `
<div>
  <div
    class="nut-receive-invoice-edit"
  >
    <div
      class="nut-receive-invoice-edit-item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            收件人
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="name"
                placeholder="请输入收件人姓名"
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nut-receive-invoice-edit-item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            手机号码
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="tel"
                placeholder="请输入收货人手机号码"
                style="text-align: left;"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nut-receive-invoice-edit-item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            所在地区
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="region"
                placeholder="请选择所在地区"
                readonly=""
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nut-receive-invoice-edit-item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            详细地址
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="address"
                placeholder="街道、楼牌号"
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="first-render hidden-render nut-overlay"
      style="z-index: 1000; animation-duration: 0.3s;"
    />
    <div
      class="nut-receive-invoice-edit-bottom"
    >
      <button
        class="nut-button nut-button--danger nut-button--normal nut-button--round nut-button--block "
      >
        <div
          class="nut-button__warp"
        >
          <div
            class=""
          >
            保存
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
